src/components/Events/Events.js

```diff
@@ -50,16 +55,32 @@ class Events extends Component {
   }
 
   componentDidMount() {
-    this.getEvents();
+    this.getPosition().then(() => this.getEvents());
+  }
+
+  getPosition() {
+    return new Promise((resolve) => {
+      if (!window.navigator.geolocation) { resolve(); }
+      window.navigator.geolocation.getCurrentPosition((pos) => {
+        this.setState({
+          geoLocation: {
+            lat: pos.coords.latitude,
+            long: pos.coords.longitude,
+            maxDistance: distanceRange[defaultRangeIndex].value,
+          }
+        });
+        resolve();
```

Comment:
Are you relying on the new state where this promised is used?  If so, you may want to call resolve as the 2nd argument to `setState()`